gc/gc-iceberg/src/main/java/org/projectnessie/gc/iceberg/IcebergContentToFiles.java

```diff
@@ -93,11 +107,26 @@ public Stream<FileReference> extractFiles(ContentReference contentReference) {
       allFiles = Stream.of(metadataUri);
     }
 
-    URI baseUri = baseUri(tableMetadata);
+    URI baseUri = checkUri("location", baseUri(tableMetadata), contentReference);
```

Comment:
I tested the latest changes. It is better now. But still, it cleans up the live data files and manifest files in the no-scheme scenario. 

The reason is we need to add `checkUri` for the `data and manifest` files also. Now it is only added for table metadata, manifest list and base location.

Reply:
This also got me worried that our tests and validations are not catching it.

Reply:
I have verified the latest fix now. It works with no scheme. Good testcases added now 👍🏻